suite/suite.go

```diff
@@ -63,6 +63,20 @@ func failOnPanic(t *testing.T) {
 	}
 }
 
+// Run provides suite functionality around golang subtests.  It should be
+// called in place of t.Run(name, func(t *testing.T)) in test suite code
+// to expose setup and teardown functionality for each subtest.  The passed-in
+// func will be executed as a subtest with a fresh instance of t.  Provides
```

Comment:
> will be executed as a subtest with a fresh instance of t.

I think this can create some confusion that a `*testing.T` instance will be passed to the passed-in function as a parameter. While obviously not the case based on the function signature of subtest for `Suite.Run`, we could probably clean up this documentation to make clear that it is the suite's testing context (as returned by `Suite.T()`) that is switched prior to execution of the given function.